File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        zero=0
        res=-1
        l=0

        for r in range(len(nums)):
            if nums[r]==0:
                zero+=1
            while zero > 1:
                if nums[l]==0:
                    zero-=1
                l+=1
            res=max(res,r-l)
        return res
        

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.java
class Solution {
    public int longestSubarray(int[] nums) {
        int res=-1;
        int zero=0;
        int l=0;
        int current=0;

        for(int r=0;r<nums.length;r++){
            if (nums[r]==0){
                zero++;
                current=r;

            }
            while(zero>1){
                if (nums[l]==0){
                    zero--;
                }
                l++;
            }
            
           
            res=Math.max(res,r-l);
        }

        return res;
        
    }
}